warehouse/warehouse/scripts/move_to_publication.py

```diff
@@ -160,7 +160,11 @@ def main():
 
     con_message("info", f"calling conduct_move with method {move_method}")
 
-    return conduct_move(parsed_args, move_method)
+    retcode = conduct_move(parsed_args, move_method)
+    if retcode != 0:
+        con_message("error", f"move_to_publication: return code = {retcode}")
```

Comment:
Does the `con_message` utility function raise a [python Error ](https://docs.python.org/3/tutorial/errors.html) with the logger message, or does it only raise a logger message?

I don't see any instances of errors being raised here: https://github.com/E3SM-Project/esgfpub/blob/711ff268e0aecf6a317baff1ad7fe37864455ea2/warehouse/warehouse/util.py#L130

It is usually best practice to raise errors whenever an unexpected/invalid condition cannot be handled in the code, rather than letting it proceed down the call stack.

More info here: https://stackoverflow.com/a/51209347
Example here: https://stackoverflow.com/a/52672315

Maybe a new issue can be opened to improve the logging/error system if it can provide high value to the debugging process? My current understanding of `esgfpub` is limited though, so you'd be better position to know if there are any benefits to this.

Reply:
In the context of warehouse state machine operation - I think we need to discuss what sorts of errors should abort processing directly, and which should return a "bad status" to the parent process.  It is not clear to me when the "clean-up" the parent should apply will, or will not be circumvented by raising a value-error.